File: modules/agent/src/models/Note.py
"""Note model for polymorphic notes across entities."""

from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from models import Base


class Note(Base):
    __tablename__ = "Note"

    id = Column(Integer, primary_key=True)
    tenant_id = Column(Integer, ForeignKey("Tenant.id", ondelete="CASCADE"), nullable=False)
    entity_type = Column(String(50), nullable=False)
    entity_id = Column(Integer, nullable=False)
    content = Column(Text, nullable=False)
    created_by = Column(Integer, ForeignKey("User.id", ondelete="SET NULL"), nullable=True)
    updated_by = Column(Integer, ForeignKey("User.id", ondelete="SET NULL"), nullable=True)
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    tenant = relationship("Tenant", back_populates="notes")
    creator = relationship("User", foreign_keys=[created_by])
    updater = relationship("User", foreign_keys=[updated_by])

    def __repr__(self):
        return f"<Note(id={self.id}, entity_type={self.entity_type}, entity_id={self.entity_id})>"

File: modules/agent/internal/models/Note.go
package models

import "time"

type Note struct {
	ID         int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	TenantID   int64     `gorm:"not null" json:"tenant_id"`
	EntityType string    `gorm:"size:50;not null" json:"entity_type"`
	EntityID   int64     `gorm:"not null" json:"entity_id"`
	Content    string    `gorm:"not null" json:"content"`
	CreatedBy  int64     `gorm:"not null" json:"created_by"`
	UpdatedBy  int64     `gorm:"not null" json:"updated_by"`
	CreatedAt  time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt  time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (Note) TableName() string {
	return "Note"
}
